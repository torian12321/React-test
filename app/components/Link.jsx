var React = require('react');

const Link = props =>(
    <a href={ props.href } className={ 'link ' + props.style }>
        { props.children }
    </a>
);

Link.propTypes = {
    children: React.PropTypes.string.isRequired,
    href    : React.PropTypes.string.isRequired,
    style   : React.PropTypes.string.isRequired
};
Link.defaultProps = {
    href : '#',
    style: ''       // arrowed | preview
};

module.exports = Link;
