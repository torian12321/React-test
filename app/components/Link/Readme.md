Basic:
```
<Link>Link to a place</Link>
```

Arrowed:
```
<div>
	<Link arrowed>Link to a place 1</Link>
	<Link arrowed>Link to a place 2</Link>
	<Link arrowed>Link to a place 3</Link>
</div>
```

Preview:
```
<Link preview src='https://www.google.com'>Google</Link>
```