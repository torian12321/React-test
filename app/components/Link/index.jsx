var
React     = require('react'),
PropTypes = require('prop-types'),
classNames= require('classnames');

const Link = props =>(
    <a href={props.href} className={classNames('link', props.className)}>
        {props.children}
    </a>
);

Link.propTypes = {
    children : PropTypes.string.isRequired,
    href     : PropTypes.string.isRequired,
    className: PropTypes.string             // arrowed | preview
};
Link.defaultProps = {
    href: '#'
};

module.exports = Link;
