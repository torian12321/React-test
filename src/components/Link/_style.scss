@import '../../styles/conf/index.scss';

// Vars
$link-color-main: $themeColor_1 !default;
$link-color-sec : $themeColor_2 !default;

// Block
.link {
	color          : $link-color-main;
	cursor         : pointer;
  text-decoration: none;

	&:hover {
		color: $link-color-sec;
	}
	&:visited {
		color: desaturate($link-color-main, 25%);
	}

  // Modifiers
	&--preview {
		position: relative;
		&::before {
			$BG: lighten($link-color-sec, 10%);

			position  : absolute;
			content   : attr(href);
			top       : 100%;
			left      : 0;
			opacity   : 0;
			padding   : .25em 1em;
			margin-top: 0.5em;
			background: $BG;
			z-index   : 1;
			font-size : 0.8em;

      @include shadow(0 0 3px 0 rgba(0,0,0,.3));
			@include smart-text-color($BG);
			@include transition();
		}
		&:hover::before {
			opacity: 1;
			left   : 24px;
		}
	}
}