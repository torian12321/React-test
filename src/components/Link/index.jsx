import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style.less';

class Link extends React.PureComponent {
  render() {
    return (
      <a 
        href     ={this.props.href}
        className={classNames(
          'link',
          {'preview': this.props.preview},
          this.props.className
        )}
      >
        {this.props.children}
      </a>
    )
  }
}

Link.propTypes = {
  children : PropTypes.string.isRequired,
  className: PropTypes.string,
  href     : PropTypes.string,
  preview  : PropTypes.bool
};
Link.defaultProps = {
  href   : '#',
  preview: false
};

module.exports = Link;
