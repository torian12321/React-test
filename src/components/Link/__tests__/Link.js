import React    from 'react';
import renderer from 'react-test-renderer';
import Link     from '../index';

describe('component', () => {
  describe('<Link />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <Link href='https://www.google.ie'>Extenal link</Link>
      )).toMatchSnapshot();
    });

    it('should render preview link', () => {
      expect(renderer.create(
        <Link href='https://www.google.ie' preview>Extenal link</Link>
      )).toMatchSnapshot();
    });

  });
});