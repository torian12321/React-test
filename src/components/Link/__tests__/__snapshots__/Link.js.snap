// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Link /> should render correctly 1`] = `
<a
  className="link"
  href="https://www.google.ie"
>
  Extenal link
</a>
`;

exports[`component <Link /> should render preview link 1`] = `
<a
  className="link preview"
  href="https://www.google.ie"
>
  Extenal link
</a>
`;
